firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigLogger.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.remoteconfig.internal;
+
+import android.util.Log;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A per-namespace logger with settable log level. Defaults to level {@link Log#INFO}.
+ *
+ * @author danasilver
+ */
+public class ConfigLogger {
+  private static final String TAG = "FirebaseRemoteConfig";
+  private static final Map<String, ConfigLogger> configNamespaceLoggers = new HashMap<>();
+
+  private final String tag;
+  private int logLevel;
+
+  private ConfigLogger(String tag) {
+    this.tag = tag;
+    this.logLevel = Log.INFO;
+  }
+
+  /** Get the logger for {@param namespace} or create one if it does not already exist. */
+  public static ConfigLogger getLogger(String namespace) {
```

Comment:
Given [RC instances are per-namespace](https://github.com/firebase/firebase-android-sdk/blob/d8f4dc14a58bfe607a455c8c984dd7327f9b6523/firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java#L48), and the logger is per-instance, I don't think we need to maintain a map of namespace-to-logger.

Reply:
I'm trying to think through whether that's necessary too... My thinking is that we need to use the same logger in the various classes injected into the instance of RC (ConfigFetchHttpClient, ConfigCacheClient, etc.) so the same instance of RC always uses the same logger, however those are created before and injected into the RC instance in [RemoteConfigComponent](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java) so they can't get the logger from the RC instance.

What do you think? I'd definitely prefer not needing to maintain the map and get the logger by namespace - would be much cleaner than passing the namespace around RemoteConfigComponent.